### DISCUSSION TITLE
Set network proxy for both IPv4 and IPv6?

### DISCUSSION THREAD

**Discussion by hg421** [sandboxie-plus/Sandboxie#4106] [Q&A Feedback] [answered]
I'm trying out the feature added in 1.14 to force sandboxed process to use a pre-defined SOCKS5 proxy.
Started a local socks server with `ssh -D 3128 (host)` and configured the box like this:
![image](https://github.com/user-attachments/assets/b874ad34-917c-45cf-8c0c-5da3acad76c3)
However it seems that only IPv4 traffic is proxied while IPv6 is not. 
If the rule is set like this
![image](https://github.com/user-attachments/assets/e62086b1-09d4-411f-adcb-567db013bf04)
then instead IPv6 is proxied but not IPv4.
Even if both rules are added, then only the rule that comes first is applied.

Hence my question: Is it possible to configure the proxy for both IPv4 and IPv6 connections?

And another side question: Is it correct that, like the firewall feature without enabled WFP, that the proxy settings are only applied in user mode and could be in theory bypassed by malicious applications?

**Comment by DavidXanatos** [comment]
you should set 2 rules one for IPv4 and one for IPv6 then it should work to proxy booth.

Yes the redirection is in user mode only, that said the you can use WFP to block traffic to any IP other then the proxy.

**Reply by hg421** [reply]
![image](https://github.com/user-attachments/assets/f2880b27-76ac-472c-bfbc-9880864cbf04)
I also tried with both rules, but it seems that only the first one takes effect... (here IPv4; if the IPv6 rule is moved up then only IPv6)

**Reply by DavidXanatos** [reply, marked as answer]
please try the sbiedll.dll from the latest CI build: https://github.com/sandboxie-plus/Sandboxie/actions/runs/10062705794

**Reply by hg421** [reply]
> please try the sbiedll.dll from the latest CI build: https://github.com/sandboxie-plus/Sandboxie/actions/runs/10062705794

Thanks, now both rules work at the same time!
Now as a suggestion, wouldn't it be more intuitive if all traffic (IPv4 and IPv6) would be proxied even if e.g. the connection to the SOCKS5 proxy is only given as IPv4? I think it should be possible because most other software I know that supports SOCKS5 works that way...
